### ISSUE TITLE
slow-compiles documentation

### ISSUE THREAD

**Issue by PeterLazar** [catchorg/Catch2#2108]
In the documentation describing slow compilation, there should be a section describing running Catch2 as a library. This is well documented in cmake-integration.md, so a simple link would suffice.

**Comment by blackliner** [comment]
Look at https://github.com/catchorg/Catch2/issues/2101, seems like Catch2 cannot be used as a (static/shared) library 😞

**Comment by PeterLazar** [comment]
I'm using it as a library.
But I did write my own main. And I think that we had to pull a development branch, because release branches are somehow very different... source files are merged into a couple of files...
Oh and that also means, that the files you include are different. So in the main.cpp I include catch_session.hpp.
In files with tests I include catch_test_macros.hpp and other files that are needed (matchers, template macros,...).

I haven't encountered any problems so far. I followed your link and saw that in another issue, the problem of registration from a static library was mentioned, but even that should be solvable by explicit registration in your main function.

And cmake-integration.md [link](https://github.com/catchorg/Catch2/blob/devel/docs/cmake-integration.md#installing-catch2-from-git-repository) does provide instructions to build it as a library. It's just that the instructions don't work on release branches.

I'm really not sure why development and release branches differ so much.
